proxy/shared/DiagsConfig.cc

```diff
@@ -281,7 +281,8 @@ DiagsConfig::DiagsConfig(const char *filename, const char *tags, const char *act
   ////////////////////////////////////////////////////////////////////
 
   if (!use_records) {
-    diags = new Diags(tags, actions, NULL);
+    diags_log = NULL;
```

Comment:
Allright, I see James' comment now. Based on that and the Coverity report I think the best option is to move that to a member initializer at the top of the constructor.